### PR TITLE
This is a collection of mostly minor (cleanup) fixes

### PR DESCRIPTION
**Pull request by hkaiser**
This is a collection of mostly minor (cleanup) fixes to various places of the threading subsystem

### CODE REVIEW

File: hpx/runtime/threads/policies/thread_queue.hpp [hunk 1]
```diff
@@ -416,27 +416,28 @@ namespace hpx { namespace threads { namespace policies {
...
                     // this thread has to be in this map
-                    HPX_ASSERT(it != thread_map_.end());
-
-                    recycle_thread(*it);
+                    HPX_ASSERT(thread_map_.find(tid) != thread_map_.end());
 
-                    thread_map_.erase(it);
+                    bool deleted = thread_map_.erase(tid) != 0;
+                    HPX_ASSERT(deleted);
+                    (void)deleted;
```
Comment (msimberg) [inline hpx/runtime/threads/policies/thread_queue.hpp]:
```suggestion
                    (HPX_UNUSED(deleted);
```

The changes in this function aren't intended to change behaviour its behaviour, right? It's not obvious to me if there's some functional change from recycling the thread last (other than allowing you to check that the `erase` is successful).

File: libs/coroutines/include/hpx/coroutines/coroutine.hpp [hunk 2]
```diff
@@ -47,6 +47,7 @@
 #include <utility>
 
 namespace hpx { namespace threads { namespace coroutines {
+
```
Comment (msimberg) [inline libs/coroutines/include/hpx/coroutines/coroutine.hpp]: Just so we don't go back and forth on formatting changes like this: I tend to remove empty lines like this :) Happy to leave them in the future if you'd prefer?

File: hpx/runtime/threads/policies/thread_queue.hpp (hunk 1 above)
Comment (hkaiser) [reply hpx/runtime/threads/policies/thread_queue.hpp]: This change was done to make it consistent with another spot in the code where we do similar things. No functional change was intended.

File: libs/coroutines/include/hpx/coroutines/coroutine.hpp (hunk 2 above)
Comment (hkaiser) [reply libs/coroutines/include/hpx/coroutines/coroutine.hpp]: Uhh. I personally prefer to have them. But I can live with it if we decided to remove them.

Comment (msimberg) [reply libs/coroutines/include/hpx/coroutines/coroutine.hpp]: No worries, let's keep it. I won't be deleting the ones that are there, but I may not always remember to add them either to new code. This is really no big deal.

**Comment by msimberg** [comment]
Just formatting left on one file (https://159071-4455628-gh.circle-artifacts.com/0/hpx/artifacts/modified_format_files.txt), then this looks pretty much good to go (at least according to CircleCI and Travis).

**Pull request merged**